src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -32,7 +33,153 @@ namespace OpenLoco::Paint
+                imageId = imageId + steamObject->var_0E + steamObject->var_0A;                                                                 // 00440364-00440367
+
+                if ((steamObject->var_08 & 8) != 0) // 0044036A-00440370
+                {
+                    session.addToPlotList2(imageId, { 0, 0, base->z }, { 1, 1, 0 }); // 00440372-00440386
+                }
+                else
+                {
+                    session.addToPlotListAsParent(imageId, { 0, 0, base->z }, { 1, 1, 0 }, { -12, -12, base->z }); // 0044038F-004403BC
+                }
+                break;
+            }
+
+            case MiscThingType::redGreenCurrency: // 1
+            {
+                if (dpi->zoom_level > 1) // 004403C5-004403D0
+                {
+                    return;
+                }
+                MoneyEffect* moneyEffect = base->asRedGreenCurrency();
+                const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_green : StringIds::format_currency_expense_red_negative; // 004403D2-004403DF
+                uint32_t currencyAmount = abs(moneyEffect->amount);
+                uint32_t yOffsets = 0x4FAAC8 + moneyEffect->wiggle; // 004403E3
+
+                session.PaintFloatingMoneyEffect(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX); // 004403F7
+                break;
+            }
+
+            case MiscThingType::windowCurrency: // 2
+            {
+                if (dpi->zoom_level > 1) // 00440400-0044040B
+                {
+                    return;
+                }
+                MoneyEffect* moneyEffect = base->asWindowCurrency();
+                const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_in_company_colour : StringIds::format_currency_expense_in_company_colour_negative; // 0044040D-0044041A
+                uint32_t currencyAmount = abs(moneyEffect->amount);
+                uint32_t yOffsets = 0x4FAAC8 + moneyEffect->wiggle;                             // 0044041E-00440422
+                uint16_t companyColour = CompanyManager::getCompanyColour(moneyEffect->var_2E); // 00440428-0044042C
+                addr<0xE3F0A8, int16_t>() = companyColour;                                      // 00440434
+
+                session.PaintFloatingMoneyEffect(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX); // 00440445
+                break;
+            }
+
+            case MiscThingType::vehicleCrashParticle: // 3
+            {
+                if (dpi->zoom_level != 0) // 0044044E-00440459
+                {
+                    return;
+                }
+
+                VehicleCrashParticle* particle = base->asVehicleCrashParticle();
+
+                constexpr uint32_t IMAGE_TYPE_REMAP = (1 << 29);
+                constexpr uint32_t IMAGE_TYPE_REMAP_2_PLUS = (1u << 31);
```

Comment:
I've created functions for this. `recolour2` applies the two flags with the colours